cal/api/views.py

```diff
@@ -242,4 +242,4 @@ class TagDetailEventTimeSeries(APIView):
 
     def get(self, request, *args, **kw):
         tag = Tag.objects.get(user=self.request.user, id=self.kwargs['pk'])
-        return Response(tag.get_time_series(self.request.query_params.get('timezone')))
+        return Response(tag.get_time_series(self.request.query_params.get('timezone'), time_step=self.kwargs['time_step']))
```

Comment:
I believeeeee using query_params requires using parameters with Http requests. Using kwargs is using regex to get the parameters from the actual url string itself. It wouldn't make sense for an url to have timezone stuff in there so you should be using query_params to get the timezone